docs/code/CrossPlatform.md

```diff
@@ -0,0 +1,157 @@
+ - LdaNative builds just fine for Arm, so no work would be required.
+ - MatrixFactorizationNative builds with the "USESSE" flag that requires Intel MKL. We can conditionally enable/disable that flag and no other work would be required.
+ - MklProxyNative can be removed/ignored for Arm builds. We will need to modify the build to exclude it as needed.
+ - SymSgdNative needs to be either re-written in managed code, or re-write 4 Intel MKL methods. The 4 methods are just dealing with vector manipulation and shouldn't be hard to do.
+
+## My Suggestion
+My suggestion would be to start with the hybrid approach. It will require the least amount of work to get ML.NET running elsewhere, while still being able to support a large majority of devices out of the gate. This solution will still limit the platforms we can run-on to Arm64 devices, but we can do a generic Arm64 compile, so it should work for all Arm64 v8 devices. It would be a good idea to have full software fallbacks so that truly anywhere .NET Core runs, ML.NET will run, such as Web Assembly and when .NET 6 comes out on mobile as well.
+
+### NET Core 5
+I think we should also add support for .NET Core 5 during this process so that we gain access to the Arm64 intrinsics. However, since .Net Core 5 goes out of support before .NET core 3.1 I don't think .NET Core 5 should be a huge focus.
+
+### 3rd Party Dependencies
+I think initially we should annotate that they don't work on non x86/x64 devices. This includes logging an error when they try and run an unsupported 3rd party dependency, and then failing gracefully with a helpful and descriptive error. The user should be able to compile the 3rd party dependency, for the ones that support it, and have ML.NET still be able to pick it up and run it if it exists. OnnxRuntime is something that we will probably want, but we can look more into this as we get requests for it in the future.
+
+### Helix
+In order to fully test everything we need to, we would also need to change how we test to use the Helix testing servers. Currently, Helix doesn't have the capability to test Apple's new M1 code, but that is in the works.
+
+### Mobile Support
+.NET Core 6 will allow us to run nativly on mobile. Since we are making these changes before .NET 6 is released, I propose we don't include that work as of yet. As long as we handle the native binaries correctly and make sure ML.NET provides descriptive error methods, we should be able to have mobile support as soon as .NET Core 6 releases for everything that currently has a software fallback.
+
+### Support Grid
+This is what I propose for the support grid. Since .NET Core 2.1 is end-of-life this year, I am putting much less emphasis on it. Since .NET Core 5 will be out of support before .NET Core 3.1 will be, I am putting less CI emphasis on .NET Core 5.
+
+| Platform | Architecture | Intel MKL | .NET Framework | .NET Core 2.1 | .NET Core 3.1 | .NET Core 5 |
```

Comment:
Which versions and distributions of the below are we going to support?

Reply:
Lets sync on this tomorrow.